### ISSUE TITLE
Missing Morrisons 

### ISSUE THREAD

**Issue by darrenaf** [pantherale0/ha-fuelprices#26]
## System Information

version | core-2024.11.0
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.12.4
os_name | Linux
os_version | 6.6.54-haos
arch | x86_64
timezone | Europe/London
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
HACS Data | ok
GitHub API Calls Remaining | 5000
Installed Version | 2.0.1
Stage | running
Available Repositories | 1461
Downloaded Repositories | 15

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 13.2
-- | --
update_channel | stable
supervisor_version | supervisor-2024.11.2
agent_version | 1.6.0
docker_version | 27.2.0
disk_total | 30.8 GB
disk_used | 8.5 GB
healthy | true
supported | true
host_connectivity | true
supervisor_connectivity | true
ntp_synchronized | true
virtualization | microsoft
board | ova
supervisor_api | ok
version_api | ok
installed_addons | Terminal & SSH (9.13.0), Studio Code Server (5.17.3), SQLite Web (4.2.2), Samba share (12.3.2), Matter Server (6.6.0), Mosquitto broker (6.4.1)

</details>

<details><summary>Dashboards</summary>

dashboards | 2
-- | --
resources | 9
views | 5
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | 29 October 2024 at 19:42
-- | --
current_recorder_run | 8 November 2024 at 10:11
estimated_db_size | 526.97 MiB
database_engine | sqlite
database_version | 3.45.3

</details>

### Checklist

- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of any [previous issues](https://github.com/pantherale0/ha-fuelprices/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

Have set up a new integration and configured my location.   Added Asda, Tesco and Morrisions, but get ESSO.  Configured to 5 miles, but MK18 and LU7 are about 12 miles away.

Asda MK1 1QB

sensor.asda_mk1_1qb ⸱ Fuel Prices ⸱ —

—

TESCO MK10 0AH

sensor.tesco_mk10_0ah ⸱ Fuel Prices ⸱ —

—

TESCO MK12 5RJ

sensor.tesco_mk12_5rj ⸱ Fuel Prices ⸱ —

—

TESCO MK1 1DD

sensor.tesco_mk1_1dd ⸱ Fuel Prices ⸱ —

—

TESCO MK18 1AB

sensor.tesco_mk18_1ab ⸱ Fuel Prices ⸱ —

—

TESCO LU7 1ER

sensor.tesco_lu7_1er ⸱ Fuel Prices ⸱ —

—

Esso MK18 1RY

sensor.esso_mk18_1ry ⸱ Fuel Prices ⸱ —

—

### Reproduction steps

1. New configuration with lat/lon set as 52.041627,-0.759651 
2.Tesco, Asda and Morrisons selected and range 5 miles
3.
...

### Debug logs

```text
Na
```

**Comment by pantherale0** [comment]
Hi,

Very odd, it seems to be working ok for me:

```
2024-11-08 15:22:25 __main__ INFO: Found location: {'id': 'asda_gcprz7m8expy', 'name': 'Asda MK1 1QB', 'address': 'Bletcham Way, Milton Keynes', 'postal_code': 'MK1 1QB', 'latitude': 52.005282, 'longitude': -0.72837, 'brand': 'Asda', 'available_fuels': {'E10': 1.297, 'B7': 1.347}, 'fuel_details': {'E10': {}, 'B7': {}}, 'currency': 'GBP', 'last_updated': '2024-11-08T15:22:08.843858', 'next_update': '2024-11-08T15:27:09.078667', 'props': {'source': 'asda', 'source_id': 'gcprz7m8expy', 'prevent_cache_cleanup': True}, 'distance': 2.8450798401474278}
2024-11-08 15:22:25 __main__ INFO: Found location: {'id': 'tesco_gcr802gw2t32', 'name': 'TESCO MK10 0AH', 'address': '1 Winchester Circle Kingston Milton Keynes', 'postal_code': 'MK10 0AH', 'latitude': 52.036471, 'longitude': -0.687309, 'brand': 'TESCO', 'available_fuels': {'E5': 1.369, 'E10': 1.2990000000000002, 'B7': 1.349}, 'fuel_details': {'E5': {}, 'E10': {}, 'B7': {}}, 'currency': 'GBP', 'last_updated': '2024-11-08T15:22:08.843858', 'next_update': '2024-11-08T15:27:09.110177', 'props': {'source': 'tesco', 'source_id': 'gcr802gw2t32', 'prevent_cache_cleanup': True}, 'distance': 3.1050058659462647}
2024-11-08 15:22:25 __main__ INFO: Found location: {'id': 'tesco_gcr2jtfmw8py', 'name': 'TESCO MK12 5RJ', 'address': 'Tesco Stores Ltd Mcconnell Drive Wolverton Milton Keynes', 'postal_code': 'MK12 5RJ', 'latitude': 52.063781, 'longitude': -0.809609, 'brand': 'TESCO', 'available_fuels': {'E5': 1.439, 'E10': 1.369, 'B7': 1.399}, 'fuel_details': {'E5': {}, 'E10': {}, 'B7': {}}, 'currency': 'GBP', 'last_updated': '2024-11-08T15:22:08.843858', 'next_update': '2024-11-08T15:27:09.110177', 'props': {'source': 'tesco', 'source_id': 'gcr2jtfmw8py', 'prevent_cache_cleanup': True}, 'distance': 2.623087914671972}
2024-11-08 15:22:25 __main__ INFO: Found location: {'id': 'tesco_gcprz6rvbj92', 'name': 'TESCO MK1 1DD', 'address': 'Tesco Stores Ltd Watling Street Bletchley Milton Keynes', 'postal_code': 'MK1 1DD', 'latitude': 52.000681, 'longitude': -0.725439, 'brand': 'TESCO', 'available_fuels': {'E5': 1.369, 'E10': 1.2990000000000002, 'B7': 1.349}, 'fuel_details': {'E5': {}, 'E10': {}, 'B7': {}}, 'currency': 'GBP', 'last_updated': '2024-11-08T15:22:08.843858', 'next_update': '2024-11-08T15:27:09.110177', 'props': {'source': 'tesco', 'source_id': 'gcprz6rvbj92', 'prevent_cache_cleanup': True}, 'distance': 3.184933834674302}
```

Can you check your settings that you have configured? Perhaps an additional area is defined that wasn't changed or removed?

Morrisons fuel stations have gone through a lot of changes this year and a new company has taken over pretty much all of them (even if the branding still remains). Currently there are now only 4 Morrisons fuel stations (that are owned and operated by Morrisons) in the whole country according to their data feed (https://www.morrisons.com/fuel-prices/fuel.json). Instead Morrisons fuel stations are now owned by MFG (https://www.motorfuelgroup.com/mfg-and-morrisons-announce-closing-of-petrol-forecourts-transaction/).

**Comment by darrenaf** [comment]
Hello

That's really odd.  This was a first time install and when I've checked there was a Home location and a MiltonKeynes (that I named) location.  I've deleted the Home and that removed the distant stations.

I've kept the Morrisons and now added the MFG group, which has brought in Morrisons locally to me.

I'm now getting ESSO, SHELL and BP, all within the 5 miles, but on checking they're part of MFG too, so that understandable.  Thanks for your assistance.
